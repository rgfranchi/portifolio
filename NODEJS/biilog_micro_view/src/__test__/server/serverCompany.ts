import { Response } from "miragejs";
import { ICompany } from "../../Company/CompanyData";
import { ICompanySlice } from "../../interfaces/ReduxInterface";

export const companyRoutes = (actions: any) => {
  let headers = {};
  // Company.
  actions.get("company/findDropdown", (schema: any) => {
    // retornar ID E NOME.
    let dataDropdown = schema.db.companies.map(
      ({ id, nome }: ICompanySlice) => ({
        id,
        nome,
      })
    );
    if (dataDropdown.length === 0) {
      return new Response(204, headers, "");
    }
    return new Response(200, headers, dataDropdown);
  });
  actions.get("company/findAll", (schema: any) => {
    const dataBase = schema.db.companies;
    if (dataBase.length === 0) {
      return new Response(204, headers, "");
    }
    return new Response(200, headers, dataBase);
  });
  actions.get("company_erro/findAll", () => {
    return new Response(500, headers, { errors: ["FAKE Server not respond"] });
  });
  actions.delete("company/:id", (schema: any, request: any) => {
    const id = request.params.id;
    if (id === "1") {
      return new Response(203, headers, { errors: ["Fake Server Delete"] });
    }
    if (id === "9999") {
      return new Response(400, headers, {
        errors: ["Fake Server Invalid Delete"],
      });
    }
    schema.companies.find(id).destroy();
    return new Response(202, headers, { errors: ["Fake Server Delete"] });
  });
  actions.get("company/findById/:id", (schema: any, request: any) => {
    if (request.params.id === "-9999") {
      return new Response(400, headers, {
        errors: ["Fake Server Error 400"],
      });
    }
    if (request.params.id === "-1") {
      return new Response(299, headers, { errors: ["Fake Server Error"] });
    }
    const company_data = schema.companies.find(request.params.id);
    if (company_data === null) {
      return new Response(204, headers);
    }
    return new Response(200, headers, company_data.attrs);
  });
  actions.post("/company", (schema: any, request: any) => {
    const attrs = JSON.parse(request.requestBody);
    delete attrs.id;
    if (attrs.logradouro === "RETORNA_ERRO_208") {
      // resposta Valida porem não esperada.
      return new Response(208, headers, "Erro FAKE");
    }
    if (attrs.logradouro === "RETORNA_ERRO_400") {
      // resposta Valida porem não esperada.
      return new Response(400, headers, "Erro FAKE 400");
    }
    if (attrs.logradouro === "RETORNA_ERRO_4XX") {
      // resposta Valida porem não esperada.
      return new Response(499, headers, "Erro FAKE 4XX");
    }

    return new Response(201, headers, schema.companies.create(attrs));
    // return schema.companies.create(attrs);
  });
  actions.put("/company", (schema: any, request: any) => {
    let attrs = JSON.parse(request.requestBody);
    if (attrs.logradouro === "RETORNA_ERRO_4XX") {
      // resposta Valida porem não esperada.
      return new Response(499, headers, "Erro FAKE 4XX");
    }
    return new Response(200, headers, schema.companies.create(attrs));
    // return schema.companies.create(attrs);
  });
};

/**
 * Valores MOCk para teste da entidade.
 * @type {import("../../Company/CompanyData").ICompany[]}
 */

export const listCompanyMock: ICompany[] = [
  {
    id: 1,
    nome: "BIILOG",
    codPostal: "04557-030",
    pais: "Brasil",
    estado: "São Paulo",
    cidade: "São Paulo",
    bairro: "Brooklin Paulista",
    logradouro: "Rua Marquês de Cascais",
    numero: "152",
    complemento: "Sala 2",
    nomeContato: "Edmondo Gazzillo",
    email: "edmondo.gazzillo@gmail.com",
    telefone1: "1132051654",
    telefone2: "11996135016",
    cnpj: "39.282.099/0001-05",
  },
  {
    id: 2,
    nome: "WINE",
    codPostal: "29168-090",
    pais: "Brasil",
    estado: "Espirito Santo",
    cidade: "Serra",
    bairro: "Civit II",
    logradouro: "Rua Comendador Alcídes Simão Helou",
    numero: "1478",
    complemento: null,
    nomeContato: "Fábio Ferreira",
    email: "fabioferreira@wine.com.br",
    telefone1: "11968574521",
    telefone2: null,
    cnpj: "03.640.467/0001-94",
  },
  {
    id: 3,
    nome: "LinconLN",
    codPostal: "07170-350",
    pais: "Brasil",
    estado: "São Paulo",
    cidade: "Guarulhos",
    bairro: "Vila Aeroporto",
    logradouro: "Av. Papa João Paulo I",
    numero: "1818",
    complemento: null,
    nomeContato: "Vander Levorato",
    email: "vander_levorato@lincolnelectric.com.br",
    telefone1: "1135284700 ",
    telefone2: null,
    cnpj: "78.803.809/0001-49",
  },
  {
    id: 4,
    nome: "CNHi",
    codPostal: "18087-220",
    pais: "Brasil",
    estado: "São Paulo",
    cidade: "Sorocaba",
    bairro: "Éden",
    logradouro: "Avenida Jerome Case",
    numero: "1801",
    complemento: null,
    nomeContato: "Não Informado",
    email: "servicos.rental@brasif.com.br",
    telefone1: "1533341700",
    telefone2: null,
    cnpj: "80.783.926/0001-30",
  },
];

// export const listCompanyMock: ICompany[] = [
//   {
//     id: 1,
//     nome: "instituto Charles Xavier UPDATE",
//     cnpj: "01.968.344/0001-51",
//     email: "xavier@xmen.hqs",
//     nomeContato: "",
//     codPostal: "05625-984",
//     pais: "Brasil",
//     cidade: "São Paulo",
//     logradouro: "Rua UPDATE",
//     bairro: "Vila Mock",
//     numero: "123",
//     complemento: "Apto A1",
//   },
//   {
//     id: 2,
//     nome: "Empresa Bootstrap A2",
//     cnpj: "07.717.737/0001-79",
//     email: "empresa@bootstrap2.com",
//     nomeContato: "Contato Bootstrap2",
//     codPostal: "05625-984",
//     pais: "Brasil",
//     cidade: "Piracicaba",
//     logradouro: "Rua .....",
//     bairro: "Vila Mock",
//     numero: "123",
//     complemento: "Apto A2",
//   },
//   {
//     id: 4,
//     nome: "Empresa Bootstrap B1",
//     cnpj: "07.717.737/0001-79",
//     email: "empresa@bootstrap.com",
//     nomeContato: "Contato Bootstrap",
//     codPostal: "05625-984",
//     pais: "Brasil",
//     cidade: "São Joaquim",
//     logradouro: "Rua .....",
//     bairro: "Vila Mock",
//     numero: "123",
//     complemento: "Apto A4",
//   },
//   {
//     id: 5,
//     nome: "Empresa Bootstrap B2",
//     cnpj: "07.717.737/0001-79",
//     email: "empresa@bootstrap2.com",
//     nomeContato: "Contato Bootstrap2",
//     codPostal: "05625-984",
//     pais: "Brasil",
//     cidade: "Leme",
//     logradouro: "Rua .....",
//     bairro: "Vila Mock",
//     numero: "123",
//     complemento: "Apto A5",
//   },
//   {
//     id: 6,
//     nome: "Empresa Bootstrap B3",
//     cnpj: "07.717.737/0001-79",
//     email: "empresa@bootstrap3.com",
//     nomeContato: "Contato Bootstrap3",
//     codPostal: "05625-984",
//     pais: "Brasil",
//     cidade: "São Roque",
//     logradouro: "Rua .....",
//     bairro: "Vila Mock",
//     numero: "123",
//     complemento: "Apto A6",
//   },
//   {
//     id: 7,
//     nome: "Empresa Bootstrap C1",
//     cnpj: "07.717.737/0001-79",
//     email: "empresa@bootstrap.com",
//     nomeContato: "Contato Bootstrap",
//     codPostal: "05625-984",
//     pais: "Brasil",
//     cidade: "Pirassununga",
//     logradouro: "Rua .....",
//     bairro: "Vila Mock",
//     numero: "123",
//     complemento: "Apto A7",
//   },
//   {
//     id: 8,
//     nome: "Empresa Bootstrap C2",
//     cnpj: "07.717.737/0001-79",
//     email: "empresa@bootstrap2.com",
//     nomeContato: "Contato Bootstrap2",
//     codPostal: "05625-984",
//     pais: "Brasil",
//     cidade: "Araras",
//     logradouro: "Rua .....",
//     bairro: "Vila Mock",
//     numero: "123",
//     complemento: "Apto A8",
//   },
//   {
//     id: 9,
//     nome: "Empresa Bootstrap C3",
//     cnpj: "07.717.737/0001-79",
//     email: "empresa@bootstrap3.com",
//     nomeContato: "Contato Bootstrap3",
//     codPostal: "05625-984",
//     pais: "Brasil",
//     cidade: "São Pedro",
//     logradouro: "Rua .....",
//     bairro: "Vila Mock",
//     numero: "123",
//     complemento: "Apto A9",
//   },
// ];

// objeto deve ter mesma ordem do teste na comparação.
/**
 * @type {import("../../Company/CompanyData").ICompany}
 */
export const companyMock: ICompany = {
  id: 5,
  nome: "BIILOG MOCK",
  codPostal: "04557-030",
  pais: "Brasil",
  estado: "São Paulo",
  cidade: "São Paulo",
  bairro: "Brooklin Paulista",
  logradouro: "Rua Marquês de Cascais",
  numero: "152",
  complemento: "Sala 2",
  nomeContato: "Edmondo Gazzillo",
  email: "edmondo.gazzillo@gmail.com",
  telefone1: "(11)3205-1654",
  telefone2: "(11)3205-1654",
  cnpj: "39.282.099/0001-05",
};

let setDelay = 100;

export let loadCompanyValidValues = async (userEvent: any, screen: any) => {
  await userEvent.type(screen.getByLabelText("Nome"), companyMock.nome);
  await userEvent.type(
    screen.getByLabelText("Contato"),
    companyMock.nomeContato
  );
  await userEvent.type(screen.getByLabelText("CNPJ"), companyMock.cnpj);
  await userEvent.type(screen.getByLabelText("E-mail"), companyMock.email);
  await userEvent.type(screen.getByLabelText("CEP"), companyMock.codPostal);
  await userEvent.type(screen.getByLabelText("Cidade"), companyMock.cidade);
  await userEvent.type(screen.getByLabelText("Bairro"), companyMock.bairro);
  await userEvent.type(screen.getByLabelText("Número"), companyMock.numero);
  await userEvent.type(
    screen.getByLabelText("Telefone Principal"),
    companyMock.telefone1
  );
  await userEvent.type(
    screen.getByLabelText("Complemento"),
    companyMock.complemento,
    {
      delay: setDelay,
    }
  );
};

export let loadCompanyInvalidValues = async (userEvent: any, screen: any) => {
  setDelay = 0;
  companyMock.nome = "AB";
  companyMock.nomeContato = "CT";
  companyMock.cnpj = "42.724.607/0001-98";
  companyMock.email = "company@xxxxx";
  companyMock.codPostal = "05625-98";
  companyMock.telefone1 = "";
  await loadCompanyValidValues(userEvent, screen);
};

export const findCompanyDropdownMock = () => {
  return listCompanyMock.map(({ id, nome }) => ({
    id,
    nome,
  }));
};
