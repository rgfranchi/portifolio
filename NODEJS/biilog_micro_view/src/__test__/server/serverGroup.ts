import { Response } from "miragejs";
import {
  ISessionAccess,
  ISessionGroup,
} from "../../interfaces/SessionInterface";
const baseURL = "group";

export const groupRoutes = (actions: any) => {
  let headers = {};
  actions.get(`${baseURL}/findAll`, (schema: any) => {
    const dataBase = schema.db.groups;
    if (dataBase.length === 0) {
      return new Response(204, headers, "");
    }
    return new Response(200, headers, dataBase);
  });
  actions.get(`${baseURL}/listGroupSelect`, (schema: any) => {
    const dataBase: any = [];
    schema.db.groups.forEach((value: any) => {
      // @ts-ignore
      dataBase.push({ id: parseInt(value.id), nome: value.nome });
    });
    if (dataBase.length === 0) {
      return new Response(204, headers, "");
    }
    return new Response(200, headers, dataBase);
  });
};

const user: ISessionAccess = {
  label: "Usuário",
  page: {
    // leitura em ...Page.tsx
    create: {
      path: "/new",
      label: "Novo Usuário",
    },
    update: {
      path: "/update",
    },
    delete: {
      path: "/delete",
    },
  },
  path: "/user", // leitura em routes.tsx
};

const company: ISessionAccess = {
  path: "/company", // leitura em routes.tsx
  label: "Empresas",
  page: {
    // leitura em ...Page.tsx
    create: {
      path: "/new",
      label: "Nova Empresa",
    },
    update: {
      path: "/update",
    },
    delete: {
      path: "/delete",
    },
  },
};

const operator: ISessionAccess = {
  path: "/operator",
  label: "Operadores",
  page: {
    // leitura em ...Page.tsx
    create: {
      path: "/new",
      label: "Inserir Operador",
    },
    update: {
      path: "/update",
    },
    delete: {
      path: "/delete",
    },
  },
};

const forklift: ISessionAccess = {
  path: "/forklift",
  label: "Equipamentos",
  page: {
    // leitura em ...Page.tsx
    create: {
      path: "/new",
      label: "Inserir Equipamento",
    },
    update: {
      path: "/update",
    },
    delete: {
      path: "/delete",
    },
  },
};

const register: ISessionAccess = {
  label: "Cadastro",
  sub_item: [
    user,
    {
      path: "/machine",
      label: "Máquina",
      page: {
        create: {
          path: "/new",
          label: "Nova Máquina",
        },
        update: {
          path: "/update",
        },
        delete: {
          path: "/delete",
        },
      },
    },
  ],
};

/**
 * Valores MOCk para teste da entidade.
 */
export const listGroupMock: ISessionGroup[] = [
  {
    id: 1,
    nome: "Root",
    access: [company, operator, register, forklift],
  },
  {
    id: 2,
    nome: "Administrador",
    access: [company, operator],
  },
  {
    id: 3,
    nome: "Usuário Comum",
    access: [operator],
  },
];
