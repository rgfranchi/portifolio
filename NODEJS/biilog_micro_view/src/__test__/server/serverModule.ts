import { IModule } from "../../Module/ModuleData";
import { loadOperatorPanelForm } from "./serverOperator";
import {
  listTransmissionMock,
  loadTransmissionValidValues,
  transmissionMock,
} from "./serverTransmission";

/**
 * Valores MOCk para teste da entidade.
 * @type {import("../../Forklift/ForkliftData").IForklift[]}
 */
export const listModuleMock: IModule[] = [
  {
    id: 1,
    codigo: "1086",
    modelo: "BII_MSO",
    fabricante: "BIILOG",
    ativo: true,
    observacoes: "WINE Modulo Antigo 238",
    companyId: 2,
    listOperatorId: [1, 2],
    placeId: 2,
    transmission: listTransmissionMock[0],
  },
  {
    id: 2,
    codigo: "1089",
    modelo: "BII_MSO",
    fabricante: "BIILOG",
    ativo: true,
    observacoes: "WINE Modulo Antigo 310",
    companyId: 2,
    listOperatorId: [1, 2],
    placeId: 2,
    transmission: listTransmissionMock[1],
  },
  {
    id: 3,
    codigo: "1165",
    modelo: "BII_MSO",
    fabricante: "BIILOG",
    ativo: true,
    observacoes: "LINCO Modulo Antigo 387",
    companyId: 3,
    listOperatorId: [4, 3],
    placeId: 3,
    transmission: listTransmissionMock[2],
  },
  {
    id: 4,
    codigo: "1167",
    modelo: "BII_MSO",
    fabricante: "BIILOG",
    ativo: true,
    observacoes: "LINCO Modulo Antigo 389",
    companyId: 3,
    listOperatorId: [4],
    placeId: 3,
    transmission: listTransmissionMock[3],
  },
  {
    id: 5,
    codigo: "1132",
    modelo: "BII_MSO",
    fabricante: "BIILOG",
    ativo: true,
    observacoes: "CNHi Modulo Antigo",
    companyId: 4,
    listOperatorId: [6, 5],
    placeId: 4,
    transmission: listTransmissionMock[4],
  },
  {
    id: 6,
    codigo: "1148",
    modelo: "BII_MSO",
    fabricante: "BIILOG",
    ativo: true,
    observacoes: "CNHi Modulo Antigo 368",
    companyId: 4,
    listOperatorId: [5],
    placeId: 4,
    transmission: listTransmissionMock[5],
  },
];

// objeto deve ter mesma ordem do teste na comparação.
export const moduleMock: IModule = {
  id: 7,
  codigo: "99ABCNEW",
  modelo: "ELÉTRICO New",
  fabricante: "BIILOG",
  ativo: true,
  observacoes: "CNHi Modulo MOCK NEW",
  companyId: 2,
  listOperatorId: [3],
  placeId: 3,
  transmission: transmissionMock,
};

// incluído delay pois na localização do erro não funciona se estiver maior que zero
let setDelay = 100;

export let loadModuleValidValues = async (
  userEvent: any,
  fireEvent: any,
  screen: any
) => {
  await userEvent.type(
    screen.getByLabelText("Código Modulo"),
    moduleMock.codigo
  );

  await loadTransmissionValidValues(userEvent, screen);
  await loadOperatorPanelForm(fireEvent, screen, moduleMock.companyId);

  await fireEvent.change(screen.getByLabelText("Localização"), {
    target: { value: moduleMock.placeId },
  });

  await userEvent.type(
    screen.getByPlaceholderText("Modelo Modulo"), //Modulo
    moduleMock.modelo
  );
  await userEvent.type(
    screen.getByPlaceholderText("Fabricante do módulo"), //Modulo
    moduleMock.fabricante
  );
  await userEvent.type(screen.getByLabelText("Ativo"), moduleMock.ativo);
  await userEvent.type(
    screen.getByPlaceholderText("Informações do módulo ..."),
    moduleMock.observacoes,
    {
      delay: setDelay,
    }
  );
};

export let loadModuleInvalidValues = async (
  userEvent: any,
  fireEvent: any,
  screen: any
) => {
  setDelay = 0;
  loadModuleInvalid();
  await loadModuleValidValues(userEvent, fireEvent, screen);
};

export let loadModuleInvalid = () => {
  moduleMock.codigo = "";
};
