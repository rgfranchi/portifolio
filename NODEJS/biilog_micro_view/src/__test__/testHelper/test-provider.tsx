import React from "react";
import { MemoryRouter, Route, Router } from "react-router-dom";

import { objMockSession } from "./test-session";

import { Provider } from "react-redux";

//componentes de tela.
import { ModalLoading } from "../../components/ModalsComponent";

/**
 * Renderiza valores para teste.
 * Utilizando o render do framework @testing-library/react
 * Componente ModalLoading.
 * @param {*} render
 * @param {*} ui
 * @param {*} history
 * @param {*} store
 * @returns
 */
export const renderProviderRouter = (
  render: any,
  ui: any,
  history: any,
  store: any,
  modal: boolean = true
) => {
  // @ts-ignore
  function Wrapper({ children }) {
    return providerRouter(children, history, store, modal);
  }
  // @ts-ignore
  return render(ui, { wrapper: Wrapper });
};

/**
 * ConstrÃ³i com base no renderProviderRouter incluindo MemoryRouter e Router.
 * @param {*} render
 * @param {*} ui
 * @param {*} history
 * @param {*} store
 * @param {*} initialEntries
 * @param {*} path
 * @returns
 */
export const renderProviderMemoryRouter = (
  render: any,
  ui: any,
  history: any,
  store: any,
  initialEntries: any,
  path: string,
  modal: boolean = true
) => {
  const wrapper = (
    <MemoryRouter initialEntries={[initialEntries]}>
      <Route path={path}>{ui}</Route>
    </MemoryRouter>
  );
  return renderProviderRouter(render, wrapper, history, store, modal);
};

/**
 * Cria as funcionalidades do Provider e Router
 * Utilizando o mount do framework Enzyme
 * @param {*} mount
 * @param {*} children
 * @param {*} history
 * @param {*} store
 * @param {boolean} modal
 * @returns
 */
export const mountProviderRouter = async (
  mount: any,
  children: any,
  history: any,
  store: any,
  modal: boolean = true
) => {
  return await mount(providerRouter(children, history, store, modal));
};

export const createSessionCurrentUser = (store: any) => {
  store.dispatch({
    type: "session/currentUser",
    payload: objMockSession,
  });
};

const providerRouter = (
  children: any,
  history: any,
  store: any,
  modal: boolean
) => {
  return (
    <Provider store={store}>
      <Router history={history}>
        {modal && <ModalLoading />}
        {children}
      </Router>
    </Provider>
  );
};
