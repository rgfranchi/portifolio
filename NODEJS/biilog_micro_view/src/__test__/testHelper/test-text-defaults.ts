import { IAlertSlice, ICompanySlice } from "../../interfaces/ReduxInterface";

export let defaultMessageTextData = {
  vaziaResponse: /vazia/i,
  falhaAcessarServidor: /Falha ao acessar servidor/i,
  idNaoDefinido: /ID não definido/i,
  criadoValor: "VALOR CRIADO XPTO",
  criadoSucesso: /Criado com sucesso/i,
  atualizadoValor: "VALOR ATUALIZADO XPTO",
  atualizadoSucesso: /Atualizado com sucesso/i,
  excluidoSucesso: "DELETED",
  erro208: "RETORNA_ERRO_208", // resposta Valida porem não esperada.
  erro208Mensagem: /INVALIDO Status/i,
  erro400: "RETORNA_ERRO_400", // resposta Valida porem não esperada.
  erro400Mensagem: /ERROR STATUS/i,
};

export let defaultFormTextValidate = {
  preenchimento: /Preenchimento obrigatório/i,
  minimo3caracteres: /com no mínimo 3 caracteres/i,
  emailInvalido: /e-mail inválido/i,
  cepFormato: /formato de 99999-999/i,
  cnpjInvalido: /CNPJ inválido/i,
  selecioneOpcao: /Selecione Uma Opção/i,
};

export let defaultPageText = {
  list: {
    access: { create: { path: "/new" }, update: {}, delete: {} },
    historyKey1: "KEY_LIST1",
    url1: "",
    historyKey2: "",
    find: ["TitleComponent", "ButtonCreate", "List"],
  },
  create: {
    access: { create: { path: "/new" }, update: {}, delete: {} },
    historyKey1: "KEY_NEW1",
    url1: "//new",
    historyKey2: "KEY_NEW2",
    find: ["TitleComponent", "CreateForm"],
  },
  update: {
    access: { create: { path: "/new" }, update: {}, delete: {} },
    historyKey1: "KEY_UPDATE1",
    url1: "//update/2",
    historyKey2: "KEY_UPDATE2",
    find: ["TitleComponent", "UpdateForm"],
  },
};

export let alertSliceDefault: IAlertSlice = {
  hidden: true,
  message: null,
  variant: "primary",
};

export let companySliceDefault: ICompanySlice = {
  id: 3,
  nome: "Company Slice Mock",
};

// @ts-ignore
export let modalSliceDefault: IConfirmModal = {
  loading: {
    show: false,
    message: null,
  },
  confirm: {
    show: false,
    executed: 0, // incrementa a cada nova tentativa.
    disabledButtons: false,
    textTitle: null,
    textBody: null,
    textAction: "Executar",
    errorExecute: null,
    variantAction: "primary",
  },
  fail: {
    show: false,
    redirect: null,
    textFail: "Falha ao Executar",
    textButton: "Falhou ... ",
  },
  success: {
    show: false,
    redirect: null,
    textSuccess: "Sucesso!!!",
    textButton: "OK",
  },
};
