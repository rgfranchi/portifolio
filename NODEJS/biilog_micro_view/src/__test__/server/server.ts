import { createServer, RestSerializer, Model } from "miragejs";
import { userRoutes } from "./serverUser";
import { listCompanyMock, companyRoutes } from "./serverCompany";
import { listOperatorMock, operatorRoutes } from "./serverOperator";
import { listUserMock } from "./serverUser";
import { listGroupMock, groupRoutes } from "./serverGroup";
import { listForkliftMock, forkliftRoutes } from "./serverForklift";
import { listPlaceMock, placeRoutes } from "./serverPlace";

let isEmpty = false;
// variÃ¡vel para resposta vazia, impede alimentar com array original.
/**
 *
 * @param {boolean} empty
 */
export const forceEmpty = (empty: boolean) => {
  // console.info("ForceEmpty", empty);
  isEmpty = empty;
};

export const getIsEmpty = () => isEmpty;

// https://miragejs.com/docs/main-concepts/database/
// https://miragejs.com/api/classes/db-collection/
// https://miragejs.com/docs/main-concepts/route-handlers/
export const mockServer = () => {
  return createServer({
    serializers: {
      reminder: RestSerializer.extend({
        include: ["list"],
        embed: true,
      }),
    },
    models: {
      // userCurrent: Model,
      // user: Model,
      user: Model,
      company: Model,
      operator: Model,
      group: Model,
      forklift: Model,
      place: Model,
    },
    seeds(server) {
      // server.db.loadData({ userCurrent: isEmpty ? [] : userCurrentMock });
      server.db.loadData({ users: isEmpty ? [] : listUserMock });
      server.db.loadData({ groups: isEmpty ? [] : listGroupMock });
      server.db.loadData({ companies: isEmpty ? [] : listCompanyMock });
      server.db.loadData({ operators: isEmpty ? [] : listOperatorMock });
      server.db.loadData({ forklifts: isEmpty ? [] : listForkliftMock });
      server.db.loadData({ places: isEmpty ? [] : listPlaceMock });
    },
    routes() {
      this.namespace = "api/v0";
      userRoutes(this);
      companyRoutes(this);
      operatorRoutes(this);
      groupRoutes(this);
      forkliftRoutes(this);
      placeRoutes(this);
      this.passthrough("https://viacep.com.br/ws/**");
    },
  });
};
