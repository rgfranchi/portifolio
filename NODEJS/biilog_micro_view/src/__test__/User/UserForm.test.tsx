import React from "react";

import { act } from "react-dom/test-utils";

import { render, fireEvent, screen, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { createMemoryHistory } from "history";

import store from "../../redux/store";

import { renderProviderRouter } from "../testHelper/test-provider";

import { defaultFormTextValidate } from "../testHelper/test-text-defaults";

import { emptyUser } from "../../User/UserData";
import {
  userMock,
  loadUserValidValues,
  loadUserInvalidValues,
} from "../server/serverUser";
import UserForm from "../../User/UserForm";
import { listGroupMock } from "../server/serverGroup";

let wrapper: any = null;
let history: any = null;
const objectMock = userMock;
const objectEmpty = emptyUser;
const submitMock = jest.fn();

beforeEach(() => {
  cleanup();
  jest.setTimeout(20000);
  history = createMemoryHistory();
});

describe("UserForm snapshot", () => {
  test("empty", async () => {
    history.entries[0].key = "EMPTY_01";
    await setWrapperDefault(objectEmpty, submitMock);
    expect(wrapper.container).toMatchSnapshot();
  });
  test("with data", async () => {
    history.entries[0].key = "DATA_01";
    await setWrapperDefault(objectMock, submitMock);
    expect(wrapper.container).toMatchSnapshot();
  });
});

describe("UserForm validate", () => {
  test("valid", async () => {
    await setWrapperDefault(objectEmpty, submitMock);
    await act(async () => {
      await loadUserValidValues(userEvent, screen);
    });
    expect(
      screen.queryByText(defaultFormTextValidate.preenchimento)
    ).toBeNull();
    await act(async () => {
      fireEvent.click(screen.getByText("Salvar"));
    });
    expect(submitMock).toHaveBeenCalledTimes(1);
  });
  test("invalid", async () => {
    await setWrapperDefault(objectEmpty, submitMock);
    await act(async () => {
      fireEvent.click(screen.getByText("Salvar"));
    });
    expect(
      screen.queryAllByText(defaultFormTextValidate.preenchimento).length
    ).toBe(4);
    await act(async () => {
      await loadUserInvalidValues(userEvent, screen);
    });
    expect(
      screen.queryAllByText(defaultFormTextValidate.minimo3caracteres).length
    ).toBe(1);
    expect(
      screen.queryAllByText(defaultFormTextValidate.selecioneOpcao).length
    ).toBe(1);
    expect(
      screen.queryAllByText(defaultFormTextValidate.emailInvalido).length
    ).toBe(1);
    expect(screen.queryAllByText(/menos uma letra caixa 'ALTA'/i).length).toBe(
      1
    );
    expect(screen.queryAllByText(/confirmação incorreto/i).length).toBe(1);
  });
});

const setWrapperDefault = async (
  object: any,
  handleSubmit: any = () => {},
  loadValues: any = { groups: listGroupMock }
) => {
  await act(async () => {
    wrapper = await renderProviderRouter(
      render,
      <UserForm
        initValues={object}
        submitForm={handleSubmit}
        loadValues={loadValues}
      />,
      history,
      store
    );
  });
};
