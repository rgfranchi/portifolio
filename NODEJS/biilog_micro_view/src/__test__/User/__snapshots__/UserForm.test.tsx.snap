// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm snapshot empty 1`] = `
<div>
  <form
    autocomplete="off"
    class=""
  >
    <div
      class="form-row"
    >
      <div
        class="form-group col-md-9"
      >
        <label
          class="form-label"
          for="ctrlNome"
        >
          Nome
        </label>
        <input
          class="form-control"
          id="ctrlNome"
          name="nome"
          placeholder="Nome do Operador"
          type="text"
          value=""
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group col-md-3"
      >
        <label
          class="form-label"
          for="ctrlGrupo"
        >
          Grupo
        </label>
        <select
          class="form-control"
          id="ctrlGrupo"
          name="groupId"
        >
          <option
            value="0"
          >
            Selecione ... 
          </option>
          <option
            value="1"
          >
            Root
          </option>
          <option
            value="2"
          >
            Administrador
          </option>
          <option
            value="3"
          >
            Usuário Comum
          </option>
        </select>
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
    <div
      class="form-row"
    >
      <div
        class="form-group col-md-4"
      >
        <label
          class="form-label"
          for="ctrlEMAIL"
        >
          E-mail
        </label>
        <input
          class="form-control"
          id="ctrlEMAIL"
          name="email"
          placeholder="email@email.com"
          value=""
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group col-md-4"
      >
        <label
          class="form-label"
          for="ctrlPassword"
        >
          Senha
        </label>
        <input
          class="form-control"
          id="ctrlPassword"
          name="password"
          placeholder="Senha do Usuário"
          type="password"
          value=""
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group col-md-4"
      >
        <label
          class="form-label"
          for="ctrlTipo"
        >
          Confirmar Senha
        </label>
        <input
          class="form-control"
          id="ctrlTipo"
          name="passwordConfirm"
          placeholder="Confirmar Senha do Usuário"
          type="password"
          value=""
        />
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
    <button
      class="btn btn-primary"
      type="submit"
    >
      Salvar
    </button>
  </form>
</div>
`;

exports[`UserForm snapshot with data 1`] = `
<div>
  <form
    autocomplete="off"
    class=""
  >
    <div
      class="form-row"
    >
      <div
        class="form-group col-md-9"
      >
        <label
          class="form-label"
          for="ctrlNome"
        >
          Nome
        </label>
        <input
          class="form-control"
          id="ctrlNome"
          name="nome"
          placeholder="Nome do Operador"
          type="text"
          value="USUÁRIO TESTE"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group col-md-3"
      >
        <label
          class="form-label"
          for="ctrlGrupo"
        >
          Grupo
        </label>
        <select
          class="form-control"
          id="ctrlGrupo"
          name="groupId"
        >
          <option
            value="0"
          >
            Selecione ... 
          </option>
          <option
            value="1"
          >
            Root
          </option>
          <option
            value="2"
          >
            Administrador
          </option>
          <option
            value="3"
          >
            Usuário Comum
          </option>
        </select>
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
    <div
      class="form-row"
    >
      <div
        class="form-group col-md-4"
      >
        <label
          class="form-label"
          for="ctrlEMAIL"
        >
          E-mail
        </label>
        <input
          class="form-control"
          id="ctrlEMAIL"
          name="email"
          placeholder="email@email.com"
          value="testemock@teste.com.br"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group col-md-4"
      >
        <label
          class="form-label"
          for="ctrlPassword"
        >
          Senha
        </label>
        <input
          class="form-control"
          id="ctrlPassword"
          name="password"
          placeholder="Senha do Usuário"
          type="password"
          value="dTVlclBAc3N3b3Jk"
        />
        <div
          class="invalid-feedback"
        />
      </div>
      <div
        class="form-group col-md-4"
      >
        <label
          class="form-label"
          for="ctrlTipo"
        >
          Confirmar Senha
        </label>
        <input
          class="form-control"
          id="ctrlTipo"
          name="passwordConfirm"
          placeholder="Confirmar Senha do Usuário"
          type="password"
          value=""
        />
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
    <button
      class="btn btn-primary"
      type="submit"
    >
      Salvar
    </button>
  </form>
</div>
`;
